plugin/src/main/java/com/denizenscript/denizen/scripts/commands/item/InventoryCommand.java

```diff
@@ -318,28 +246,52 @@ else if (type == InventoryType.WORKBENCH) {
-        InventoryTag destination = destinationentry.getValue();
-        ElementTag slot = scriptEntry.getElement("slot");
-        ElementTag mechanism = scriptEntry.getElement("mechanism");
-        ObjectTag mechanismValue = scriptEntry.getObjectTag("mechanism_value");
-        DataAction flagAction = (DataAction) scriptEntry.getObject("flag_action");
-        TimeTag expiration = scriptEntry.getObjectTag("expiration");
-        if (scriptEntry.dbCallShouldDebug()) {
-            Debug.report(scriptEntry, getName(), db("actions", actions), destination, origin, mechanism, mechanismValue, flagAction, expiration, slot);
+    public static void autoExecute(ScriptEntry scriptEntry,
+                                   @ArgName("actions") @ArgLinear @ArgSubType(Action.class) List<Action> actions,
+                                   @ArgName("origin") @ArgRaw @ArgPrefixed @ArgDefaultNull String originString,
+                                   @ArgName("destination") @ArgRaw @ArgPrefixed @ArgDefaultNull String destinationString,
+                                   @ArgName("slot") @ArgPrefixed @ArgDefaultNull String slot,
+                                   @ArgName("expiration") @ArgPrefixed @ArgDefaultNull ObjectTag expiration,
+                                   @ArgName("data_action") @ArgRaw @ArgLinear @ArgDefaultNull String dataAction) {
+        if (slot == null) {
+            if (actions.contains(Action.ADJUST)) {
+                Debug.echoError("Inventory adjust must have an explicit slot!");
+            }
+            else {
+                slot = "1";
+            }
         }
-        int slotId = SlotHelper.nameToIndexFor(slot.asString(), destination.getInventory().getHolder());
+        TimeTag expire = null;
+        if (actions.contains(Action.FLAG) && expiration != null) {
+            if (expiration.canBeType(DurationTag.class)) {
+                TimeTag now = TimeTag.now();
+                expire = new TimeTag(now.millis() + expiration.asType(DurationTag.class, scriptEntry.context).getMillis(), now.instant.getZone());
+            }
+            else if (expiration.canBeType(TimeTag.class)) {
+                expire = expiration.asType(TimeTag.class, scriptEntry.context);
+            }
+            else {
+                Debug.echoError("Flag expiration is not a DurationTag or TimeTag!");
+            }
+        }
+        AbstractMap.SimpleEntry<Integer, InventoryTag> originEntry = originString != null ? Conversion.getInventory(new Argument(originString), scriptEntry) : null;
+        AbstractMap.SimpleEntry<Integer, InventoryTag> destinationEntry = destinationString != null ? Conversion.getInventory(new Argument(destinationString), scriptEntry) : null;
+        if (destinationEntry == null) {
+            InventoryTag inv = Utilities.getEntryPlayer(scriptEntry).getInventory();
+            if (inv == null) {
+                Debug.echoError("Must specify a Destination Inventory!");

```

Comment:
These other errors still need to be swapped over to `InvalidArgumentsRuntimeException`